pkg/controller/replicaset/replica_set.go

```diff
@@ -580,6 +580,15 @@ func (rsc *ReplicaSetController) syncReplicaSet(key string) error {
 		return err
 	}
 
+	//Do a full fetch here.
```

Comment:
You might as well replace the call to the lister.

nit: space between // and comment.

Reply:
Wait, I see this is detecting a deletion, so both calls are necessary.

Reply:
The opposite race can also happen, where the cache says "not found" even though it's just been added. I think the only consequence to that at the moment is a delay in processing until something else happens to the RS object.